executor/insert_test.go

```diff
@@ -480,6 +480,38 @@ func (s *testSuite3) TestInsertZeroYear(c *C) {
 	))
 }
 
+func (s *testSuite3) TestInsertInvalidYear(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec(`drop table if exists t1;`)
+	tk.MustExec(`create table t1(y YEAR);`)
+
+	tk.MustExec(`set sql_mode = 'STRICT_TRANS_TABLES';`)
+	_, err := tk.Exec(`insert into t1 values(1900), (2156), ("1900"), ("2156");`)
+	c.Assert(err.Error(), Equals, `[types:8033]invalid year`)
+
+	_, err = tk.Exec(`insert ignore into t1 values(1900), (2156), ("1900"), ("2156");`)
+	c.Assert(err, IsNil)
+	tk.MustQuery("show warnings;").Check(testutil.RowsWithSep("|",
+		"Warning|8033|invalid year",
```

Comment:
This warning is different with `MySQL-8.0.23`, `MySQL-5.7.33`, `TiDB-5.7.25-TiDB-v4.0.11`. They all will produce this error `(1264, "Out of range value for column 'y' at row 1")`.
```
show warnings;
+---------+------+--------------------------------------------+
| Level   | Code | Message                                    |
+---------+------+--------------------------------------------+
| Warning | 1264 | Out of range value for column 'y' at row 1 |
| Warning | 1264 | Out of range value for column 'y' at row 2 |
| Warning | 1264 | Out of range value for column 'y' at row 3 |
| Warning | 1264 | Out of range value for column 'y' at row 4 |
+---------+------+--------------------------------------------+
```

Reply:
I noticed this bug, but if I modified it, there will be some test cases that fail to pass, we'd better raise a new issue separately to deal with this problem.

Reply:
OK. Could you please create a bug issue and add a comment about it in this PR?

Reply:
ok